src/tiledbsoma_ml/gpu_batches.py

```diff
@@ -0,0 +1,90 @@
+# Copyright (c) 2021-2024 The Chan Zuckerberg Initiative Foundation
+# Copyright (c) 2021-2024 TileDB, Inc.
+#
+# Licensed under the MIT License.
+from __future__ import annotations
+
+import logging
+from typing import Iterable, Iterator
+
+import attrs
+import numpy as np
+import pandas as pd
+from scipy import sparse
+
+from tiledbsoma_ml._utils import EagerIterator
+from tiledbsoma_ml.common import Batch
+from tiledbsoma_ml.io_batches import IOBatches
+
+logger = logging.getLogger("tiledbsoma_ml.gpu_batches")
+
+
+@attrs.define(frozen=True)
+class GPUBatches(Iterable[Batch]):
```

Comment:
Can you elaborate on the benefit of migrating from generator functions to Iterable classes?  I am curious if this is just a style preference, or if there is an actual benefit.  I intentionally migrated to generators to provide more explicit encapsulation of state (some of which is illusory - a sufficiently motivated user could work around much of it), and because generators felt much less verbose in simple cases (i.e., didn't create new types, etc).

this is a curiosity question - both will work fine.

Reply:
TY - appreciate the insights.